test/parallel/test-global-buffer-dep.mjs

```diff
@@ -0,0 +1,19 @@
+// Flags: --experimental-modules
+import '../common';
+import assert from 'assert';
+import process from 'process';
+
+let sawWarning = false;
+process.on('warning', (warning) => {
+  assert.strictEqual(warning.name, 'DeprecationWarning');
+  assert.strictEqual(warning.message,
+                     'global.Buffer is deprecated in ECMAScript Modules. ' +
+                     'Please use `import { Buffer } from \'buffer\'` instead.');
+  sawWarning = true;
+});
+
+global.Buffer;
+
+setImmediate(() => {
+  assert.strictEqual(sawWarning, true);
+});
```

Comment:
Please use `common.expectWarning()` instead of the manual handling here.

Reply:
I've added back a deprecation code and used this API now instead. Previously this was done to remove the deprecation code as this was not an official deprecation of a legacy path.